text/0000-posix-error-numbers.md

```diff
@@ -0,0 +1,248 @@
+- Feature Name: posix_error_numbers
+- Start Date: 2020-08-09
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a `std::os::unix::ErrorNumber` struct for storing and comparing POSIX error
+numbers.
+
+# Motivation
+[motivation]: #motivation
+
+The POSIX standard defines a set of named [error numbers], such as `EIO` or
+`EPIPE`, for use in communicating error conditions from low-level OS routines.
+Many of these are currently exposed in [`libc`] as `c_int` constants, because
+they figure prominently in writing bindings to libc on UNIX platforms.
```

Comment:
The struct makes it easier to namespace the error number symbols. This could be done in a module if you'd prefer, like `std::os::unix::error_numbers`, but I think that would be less ergonomic.